### PR TITLE
[DOC] Add a raptor search tutorial

### PR DESCRIPTION
**Pull request by Irallia**

### CODE REVIEW

File: doc/tutorial/03_search/index.md [hunk 1]
```diff
@@ -8,8 +8,315 @@ You will learn how to searche a Raptor index using one or more sequences queries
...
+
+\note
+**K-mer counting lemma**: For a given `k` and number of errors `e`, there are \f$k_p = |p| - k + 1\f$ many k-mers in the
+pattern `p` and an approximate occurrence of `p` in text `T` has to share at least \f$t = (k_p - k \cdot e)\f$ k-mers. \n
+**Example:** For the query ACGTT und `k=2` we have \f$(5 - 2 + 1) = 4\f$ k-mers (AC CG GT TT). With an error `= 1` we
+have \f$4 - (2 \cdot 1) = 2\f$ k-mers which have to exist for matching the query. \n
+Alternative: Given a threshold of `50%` \f$0.5 \cdot 4 = 2\f$ k-mers have to exist in a bin for matching the query.
+
+\assignment{Assignment 2: Search with an error or a threshold.}
+From the first assignment in the last tutorial we want to use the more usefull `raptor2.index` and do two searches on it.
```
Comment (Irallia) [inline doc/tutorial/03_search/index.md:149]: Use minimiser.index

File: doc/tutorial/03_search/index.md [hunk 2]
```diff
@@ -8,8 +8,315 @@ You will learn how to searche a Raptor index using one or more sequences queries
...
+Raptor also supports parallelisation in search. By specifying `--threads`, for example, the querys are processed
+simultaneously.
+
+### Time
+
+If you want to know the duration of raptor search including *time for reading the index*, *time for reading reads* and
+*the computation time* for benchmarking, you can use `--time` to write a timing file.
+
+### HIBF
+
```
Comment (Irallia) [inline doc/tutorial/03_search/index.md:313]: add a hibf assignment with using hibf2.index?

File: doc/tutorial/04_search/index.md [hunk 3]
```diff
@@ -0,0 +1,339 @@
+# Search with Raptor {#tutorial_search}
+
+You will learn how to searche a Raptor index using one or more sequences queries.
```
Comment (eseiler) [inline doc/tutorial/04_search/index.md:3]:
```suggestion
You will learn how to search a Raptor index using one or more sequences queries.
```

File: doc/tutorial/04_search/index.md [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+
+\tutorial_head{Easy, 30 min, \ref tutorial_index, }
+
+[TOC]
+
+# Search an index using one or more sequences queries
+
+## A first search
+
+Now we want to have a look at Raptors second method: `raptor search`.
```
Comment (eseiler) [inline doc/tutorial/04_search/index.md:13]:
```suggestion
Now we want to have a look at Raptor's second method: `raptor search`.
```

File: doc/tutorial/04_search/index.md [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+
+We have already learned how to build an index in the previous tutorial on the `raptor index` and now we want to learn
+how to search in it.
+But before we delve deeper into the various possibilities and parameters. We can start a first search with the indexes
```
Comment (eseiler) [inline doc/tutorial/04_search/index.md:17]:
```suggestion
But before that, we delve deeper into the various possibilities and parameters. We can start a first search with the indexes
```

File: doc/tutorial/04_search/index.md [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+After the header section, each line of the output consists of the read ID (in this example these are numbers) and the
+corresponding bins in which they were found.
+
+\assignment{Assignment 1: Create a first search on our created example files and their raptor index}
+From the first assignment in the last tutorial your directory should look like this:
+```bash
+tmp$ ls
+all_paths.txt   mini_1.fasta    mini_2.fasta    mini_3.fasta    raptor.index    raptor2.index    ...
+```
+Lets search for the queries `CGCGTTCATT` amd `CGCGTCATT` with the first two indexes, with creating a `search.output` and
```
Comment (eseiler) [inline doc/tutorial/04_search/index.md:55]: The query is shorter than k (19 for the first index, 4 for the second).
I don't check this (you usually search long sequences), but it needs to be >=k.

File: doc/tutorial/04_search/index.md [hunk 7]
```diff
@@ -0,0 +1,339 @@
...
+
+\note
+The Raptor search also has a help page, which can be accessed as usual by typing `raptor search -h` or
+`raptor search --help`. And also an advanced help page: `raptor search -hh` or `raptor search --advanced-help`.
+
+## General Idea & main parameters
+
+\image html Raptor_no_title.svg width=90%
+
+\todo wollen wir die Raptor überschrift im image abschneiden?
```
Comment (eseiler) [inline doc/tutorial/04_search/index.md:104]: Done?

File: doc/tutorial/04_search/index.md [hunk 8]
```diff
@@ -0,0 +1,339 @@
...
+#2      mini_3.fasta
+#QUERY_NAME     USER_BINS
+query1  0
+query2
+```
+In the following section, we will also discuss these results and their differences.
+\endsolution
+
+\todo
+Warum hat query2 keinen match?
```
Comment (eseiler) [inline doc/tutorial/04_search/index.md:94]: See other comment about `k`

**Review by eseiler — CHANGES_REQUESTED** [review]

**Review by eseiler — APPROVED** [review]

**Pull request merged**